app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt

```diff
@@ -0,0 +1,109 @@
+import com.wire.android.R
+import com.wire.android.ui.common.NavigationIconType
+import com.wire.android.ui.common.topappbar.AppTopBarWithSearchBarLayout
+import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.home.newconversation.contacts.ContactsScreen
+import com.wire.android.ui.home.newconversation.search.SearchPeopleScreen
+import com.wire.android.ui.home.newconversation.search.SearchPeopleViewModel
+
+
+@Composable
+fun NewConversationRouter(newConversationViewModel: NewConversationViewModel = hiltViewModel()) {
+    val newConversationState = rememberNewConversationState()
+
+    AppTopBarWithSearchBarLayout(
+        scrollPosition = newConversationState.scrollPosition,
+        searchBarHint = stringResource(R.string.label_search_people),
+        searchQuery = newConversationState.searchQuery,
+        onSearchQueryChanged = {
+            newConversationState.searchQuery = it
+        },
+        onSearchClicked = { newConversationState.navigateToSearch() },
+        onCloseSearchClicked = {
+            newConversationState.clearSearchQuery()
+            newConversationState.navigateBack()
+        },
+        appTopBar = {
+            WireCenterAlignedTopAppBar(
+                elevation = 0.dp,
+                title = stringResource(R.string.label_new_conversation),
+                navigationIconType = NavigationIconType.Close,
+                onNavigationPressed = { newConversationViewModel.close() }
+            )
+        },
+        content = {
+            NavHost(newConversationState.navController, startDestination = "contacts") {
+                composable(
+                    route = "contacts",
+                    content = {
+                        ContactsScreen(onScrollPositionChanged = { newConversationState.updateScrollPosition(it) })
+                    }
+                )
+                composable(
+                    route = "search_people",
+                    content = {
+                        val searchPeopleViewModel: SearchPeopleViewModel = hiltViewModel()
+
+                        LaunchedEffect(newConversationState.searchQuery) {
+                            searchPeopleViewModel.search(newConversationState.searchQuery)
+                        }
+
+                        SearchPeopleScreen(
+                            searchPeopleState = searchPeopleViewModel.state,
+                        )
+                    }
+                )
+            }
+        }
+    )
+}
+
+class NewConversationStateScreen(
+    val navController: NavHostController
+) {
+
+    var searchQuery by mutableStateOf("")
+
+    var scrollPosition by mutableStateOf(0)
+        private set
+
+    fun updateScrollPosition(newScrollPosition: Int) {
+        scrollPosition = newScrollPosition
+    }
+
+    fun navigateToSearch() {
+        navController.navigate("search_people")
+    }
+
+    fun navigateBack() {
+        navController.navigate("contacts")
```

Comment:
I know this screen is a bit out of scope but can we at least extract these navigation operations to a const val or something similar and leave up to the `NewConversationViewModel` the navigation responsibility ?